etc/kubernetes/helm/enterprise-gateway/values.yaml

```diff
@@ -63,10 +63,13 @@ deployment:
   affinity: {}
   nodeSelector: {}
   extraEnv: {}
-    # EG_DEFAULT_KERNEL_MANAGER_CLASS: enterprise_gateway.services.kernels.remotemanager.YourOwnKernelManager
-    # SOME_ENV_VAR_WITH_LONG_VALUE: |-
-    #   Split it like this
-    #   For better readability
+    # EG_LIST_KERNELS: False
+    # EG_ALLOWED_KERNELS: >
```

Comment:
I thought this was an excellent example, then recalled that this will conflict with `kernel.allowedKernels` below.

Hmm, I also see `kernel.cullConnected` is defined as well - so we seem to have a number of helm variables that (ultimately) manifest themselves as envs in [`deployment.yaml`](https://github.com/jupyter-server/enterprise_gateway/blob/eb88e1b44a68e870c025c9a806d061db3394fa72/etc/kubernetes/helm/enterprise-gateway/templates/deployment.yaml#L49).

Note: There is a bug, however, in that there's no mapping in `deployment.yaml` from `.Values.kernel.cullConnected` to `EG_CULL_CONNECTED`.   Could you please include a fix for that in this PR?

How about using descriptive names like in your first commits, something like `SOME_ENV_VAR_WITH_SIMPLE_VALUE` and `SOME_ENV_VAR_WITH_LONG_VALUE`?

I really appreciate that you included both kinds of examples btw - thank you!

Reply:
>> There is a bug, however, in that there's no mapping in deployment.yaml from .Values.kernel.cullConnected to EG_CULL_CONNECTED.

This was an oversight on my part actually.  I forgot that you had connected the dots already.

> In my opinion, it would be best to let the users manipulate all the EG related environment variables by themselves via a meta way like extraEnv and focus on exposing the list of possible environment variables 

I agree with this and wonder if we should refactor the helm scripts a bit to address this in the 4.0 timeframe.  I suppose this could be done prior to a major release if we found that necessary.